cg/test/management/stats.api.spec.js

```diff
@@ -0,0 +1,107 @@
+jest.mock('../../src/utils/blockchain/web3-provider-factory');
+
+const winston = require('winston');
+
+const { initResources, fetch, closeResources } = require('../utils');
+const { deployContract } = require('../blockchain-setup');
+const { db } = require('../../src/db');
+const { managementJWT, subjectJWT } = require('../../src/utils/jwt');
+const { setProcessors } = require('../../src/utils/blockchain');
+
+const { VALID_RUN_MODES } = require('../../src/utils/constants');
+
+const processor1Address = '0x00000000000000000000000000000000000000A1';
+
+beforeAll(async () => {
+  try {
+    await deployContract();
+  } catch (e) {
+    winston.error(`Failed deploying contract ${e.toString()}`);
+  }
+  await initResources();
+
+  let processorsToInsert = [{ id: 201, name: 'Processor 1' }];
+  await db('processors').insert(processorsToInsert);
+
+  const processors = await db.select('id', 'name').from('processors');
+  processorsToInsert.forEach(p => expect(processors).toContainEqual(p));
+
+  let addressesToInsert = [{ processor_id: 201, address: processor1Address }];
+
+  await setProcessors(addressesToInsert.map(a => a.address));
+
+  await db('processor_address').insert(addressesToInsert);
+
+  const addresses = await db('processor_address');
+  addressesToInsert.forEach(a => expect(addresses).toContainEqual(a));
+});
+
+beforeEach(() => {
+  process.env.MODE = VALID_RUN_MODES.CONTROLLER;
+});
+afterAll(closeResources);
+
+describe('Stats endpoint', () => {
+  it('Should display the stats for the current system state', async () => {
```

Comment:
I suggest a test case (at least one) with no data first to see if we get zero's. I suspect a 500 error when there are no subjects. Maybe also some edge cases when consented is 0 and total is > 0, etc.